vaadin-cdi/src/test/java/com/vaadin/cdi/context/RouteContextualStorageManagerTest.java

```diff
@@ -243,6 +249,38 @@ public void onBeforeEnter_anotherUIHasNoWindowName_beanIsDestroyedOnUiDestroy()
         Assert.assertNotEquals(STATE, memberOfGroup1.get().getState());
     }
 
+    @Test
+    public void getActiveContextualStorages_getAllStoragesForGivenUI() {
+        RouteScopedContext context = new RouteScopedContext(manager);
+        context.init(manager, null);
+
+        UI ui = doSetUp("foo", null);
+
+        Mockito.when(event.getNavigationTarget())
+                .thenReturn((Class) Group1.class);
+        beforeNavigationTrigger.fire(event);
+
+        MemberOfGroup1 bean = memberOfGroup1.get();
+
+        List<ContextualStorage> storages = context
+                .getActiveContextualStorages();
+        Assert.assertEquals(1, storages.size());
+        ContextualStorage storage = storages.get(0);
+
+        doSetUp(null, ui.getSession());
+
+        Mockito.when(event.getNavigationTarget())
+                .thenReturn((Class) Group1.class);
+        beforeNavigationTrigger.fire(event);
+
+        MemberOfGroup1 bean1 = memberOfGroup1.get();
+
+        storages = context.getActiveContextualStorages();
+        Assert.assertEquals(1, storages.size());
+        // Now the second UI is active and its storage is a different one
```

Comment:
I don't get the meaning of prerequisites for this assertion.
`UI ui = doSetUp("foo", null);` creates a new UI with corresponding window name  "foo", adds it to the session and sets it as active.
`doSetUp(null, ui.getSession());` creates another UI with the same window name as previous and sets it as active, the previous one is not active anymore.
Then you check that the active storage being returned by context corresponds to the second UI.
Is that right?

Reply:
`doSetUp(null, ui.getSession())` creates _another_ UI with not defined window name: because it's not specified and mock which returns window name is not called.

The main purpose here is : create TWO different UIs.
`doSetUp()` activates the last created UI: it makes it active via set `UI:setCurrent`. But this is an impl details.
`doSetUp` makes active the `UI` which it creates. As a result there are two `UI`s and only one is active.
And test checks exactly what's implemented in the code: that only storages of the second UI are returned when method `getActiveContextualStorages` is called.
The storage of the first `UI` ( which is inactive) is not returned.
So prerequisites are : two different UIs.